spec/src/main/doc/javaee/injectionandresolution_ee.asciidoc

```diff
@@ -0,0 +1,104 @@
+* the bean is available for injection in the war containing the JSP or JSF page with the EL expression.
+
+[[ambig_names_ee]]
+
+==== Ambiguous EL names
+
+When running in Java EE, the container must extend the rules defined in <<ambig_names>> to names used in Expression Language.
+
+[[injection_ee]]
+
+=== Dependency injection in Java EE
+
+When running in Java EE, the container must extend the rules defined in <<injection>> and is also required to perform dependency injection whenever it creates the following contextual objects:
+
+* contextual instances of EJB session beans.
+
+The container is also required to perform dependency injection whenever it instantiates any of the following non-contextual objects:
+
+* non-contextual instances of EJB session beans (for example, session beans obtained by the application from JNDI or injected using +@EJB+), and
+* instances of any other Java EE component class supporting injection.
+
+A Java EE 5 container is not required to support injection for non-contextual objects.
+
+[[instantiation_ee]]
+
+==== Injection using the bean constructor in Java EE
+
+When running in Java EE, the container must extend the rules defined for managed beans in <<instantiation>> to EJB session beans.
+
+[[fields_initializer_methods_ee]]
+
+==== Injection of fields and initializer methods in Java EE
+
+When running in Java EE, the container must extend the rules defined for managed beans in <<fields_initializer_methods>> to EJB session beans and to any other Java EE component class supporting injection.
+
+The container is also required to ensure that:
+
+-* Initializer methods declared by a class X in the type hierarchy of the bean are called after all Java EE component environment resource dependencies declared by X or by superclasses of X have been injected.
+-* Any +@PostConstruct+ callback declared by a class X in the type hierarchy of the bean is called after all Java EE component environment resource dependencies declared by X or by superclasses of X have been injected.
+-* Any servlet +init()+ method is called after all initializer methods have been called, all injected fields have been initialized and all Java EE component environment resource dependencies have been injected.
+
+[[dependent_objects_destruction_ee]]
+
+==== Destruction of dependent objects in Java EE
+
+When running in Java EE, the container must extend the rules defined for managed beans in <<dependent_objects_destruction>> to any other Java EE component class supporting injection.
+
+[[injection_point_ee]]
+
+==== Injection point metadata in Java EE
+
+When running in Java EE, the container must extend the rules defined in <<injection_point>> and is also required to ensure that:
+
+If a Java EE component class supporting injection that is not a bean has an injection point of type +InjectionPoint+ and qualifier +@Default+, the container automatically detects the problem and treats it as a definition error.
```

Comment:
Yes but nearly all these contraints are checked at boot time (automatic). We have specific rules for `Instance#get` since constraint check are done at runtime but nothing . So I think I should open a ticket to specify how contraints should be treated for unmanaged objects. Wdyt?